warehouse/query-core/src/main/java/datawave/query/util/AttributeComparator.java

```diff
@@ -0,0 +1,203 @@
+package datawave.query.util;
+
+import com.google.common.collect.Sets;
+import datawave.query.attributes.Attribute;
+import datawave.query.attributes.Attributes;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public final class AttributeComparator {
```

Comment:
Would like to see class level docs that explain the following..
- why are we comparing attributes in the first place?
- what determines when two attributes equivalent
- perhaps add a simple example

Reply:
Added in recent commit